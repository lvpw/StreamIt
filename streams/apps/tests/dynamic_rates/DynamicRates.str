void->void pipeline DynamicRates 
{
    add Filter1();
    add Filter2();
    add Filter3();
}

void->int filter Filter1() 
{
    int i = 0;
    
    work push 1 {
	if (i == 10) 
	    i = 0;
	push(i++);
    }
}


int->int filter Filter2() 
{
    work push 1 pop [1,*] {
	int val, i, tot = 0;
	
	val = pop();
	
	for (i = 0; i < val; i++) {
	    tot += pop();
	}
	
	push(tot);
    }
}

int->void filter Filter3() 
{
    work pop 1 
	{
	    println(pop());
	}
}

