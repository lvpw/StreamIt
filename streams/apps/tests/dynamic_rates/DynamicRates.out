0
2
15
27
30
2
15
27
30
2
